google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -1900,7 +1925,7 @@ public Builder toBuilder() {
 
   @Override
   public int hashCode() {
-    return Objects.hash(name);
+    return Objects.hash(bucketWithProject.getBucketName());
```

Comment:
I don't think we should reach down into the bucketWithProject here. That could result in having two buckets with differing projects but the same name generating the same hashCode.